src/chain/tezos/TezosNodeWriter.ts

```diff
@@ -957,4 +957,44 @@ export namespace TezosNodeWriter {
 
         return ''; // applied
     }
+
+    /**
+     * Get the result of a smart contract view using the Tezos node's `run_script_view` RPC.
+     *
+     * @param {string} server Tezos node to connect to
+     * @param {string} chainid The chain ID to apply the operation on, should be "main" or "test", not to be confused with chain_id.
+     * @param {TezosP2PMessageTypes.Operation} operations A set of operations to update.
+     * @param {string} contract_address The address of the view's contract.
+     * @param {string} view_name The view to run
+     * @param {string} view_input The input to provide to the view
+     * @param {string} chain_id The chain id hash, not to be confused with chainid. Default is mainnet chain_id, NetXdQprcVkpaWU.
+     * @returns {Promise<any>} JSON-encoded response
+     */
+    export async function runView(
+        server: string,
+        chainid: string,
+        contract_address: string,
+        view_name: string,
+        view_input: any,
+        chain_id: string = 'NetXdQprcVkpaWU',
```

Comment:
the chainID var is there twice

Reply:
also we probably don't need the chain id as I see in all other functions we always use chainid as "main" there is no other value being used even for gas estimation or sending transactions. tagging @anonymoussprocket to confirm that's the case

Reply:
need block number as a param

Reply:
I preserved the two chain ID parameters to keep the new function consistent with the old ones. To change this, we should edit all functions in a separate PR. 

Block number has been added.